tests/release/release.go

```diff
@@ -134,4 +141,132 @@ var _ = framework.ReleaseSuiteDescribe("test-demo", func() {
+				a Release test-release in the matching-scenario-user namespace and failed;
+				The following fields are set as follows:
+				Status="False"
+				REASON="ReleaseValidationError"
+				Message="no ReleaseLink found in target workspace 'matching-scenario-managed' with target 'matching-scenario-user' and application 'test'"
+		*/
+
+		var _ = Describe("Failure test #1 - create resources", func() {
+			It("Create a an ApplicationSnapshot for M5 failure#1 application", func() {
+				_, err := framework.ReleaseController.CreateApplicationSnapshot(snapshotName, Failure1SourceNamespace, Failure1ApplicationName, snapshotComponents)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Create Release Link in failure#1 source namespace", func() {
+				_, err := framework.ReleaseController.CreateReleaseLink(Failure1SourceReleaseLinkName, Failure1SourceNamespace, Failure1ApplicationName, Failure1ManagedNamespace, "")
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Create a Release in failure#1 source namespace", func() {
+				_, err := framework.ReleaseController.CreateRelease(Failure1ReleaseName, Failure1SourceNamespace, snapshotName, Failure1SourceReleaseLinkName)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+		})
+
+		// Verification of test failures
+		/* What to expect -
+
+		After running the commands above, you should expect to find:
+		a ReleaseLink test-releaselink in the matching-scenario-user namespace;
+		a Release test-release in the matching-scenario-user namespace and failed;
+		the REASON field of the test-release Release set to Error.
+
+		*/
+		var _ = Describe("Failure test #1 Verification", func() {
+
+			release := &v1alpha1.Release{}
+			releaseReason := ""
+			releaseMessage := ""
+
+			// Check if there is a ReleaseLink in managed namespace
+			It("The ReleaseLink test-releaselink has been created in the failure#1 source namespace ", func() {
+				Eventually(func() bool {
+					releaseLink, err := framework.ReleaseController.GetReleaseLink(Failure1SourceReleaseLinkName, Failure1SourceNamespace)
+
+					return releaseLink != nil && err == nil
+				}, avgPipelineCompletionTime, defaultInterval).Should(BeTrue())
+			})
+
+			It("The Release have been created and failed with the REASON field set to ReleaseValidationError", func() {
+				Eventually(func() bool {
+					release, err = framework.ReleaseController.GetRelease(Failure1ReleaseName, Failure1SourceNamespace)
+
+					if err != nil {
+						return false
+					}
+
+					if release != nil {
+						releaseReason = release.Status.Conditions[0].Reason
+					} else {
+						return false
+					}
+
+					return release.IsDone() && meta.IsStatusConditionFalse(release.Status.Conditions, "Succeeded") && Expect(releaseReason).To(Equal("ReleaseValidationError"))
+				}, avgPipelineCompletionTime, defaultInterval).Should(BeTrue())
+			})
+
+			It("The Release have failed because no ReleaseLink found in target workspace", func() {
```

Comment:
```suggestion
			It("Condition message describes an error finding a matching ReleaseLink", func() {
```

Reply:
I'd use an eventually block here as well. Imagine that you remove the test above. You just added a race condition.